src/lib/dom.js

```diff
@@ -741,7 +741,7 @@ export function isInline(elm, includeCodeAsBlock) {
 
 	tagName = elm.tagName.toLowerCase();
 
-	if (tagName === 'code') {
+	if (tagName === 'code' && !is(elm, '.inline')) {
```

Comment:
I have some doubts about this.... But I do understand it is a hard situation you were placed in due to the previous code....
I'll leave such decision to @samclarke

Reply:
I don't like that by myself, as well as the hardcoded enumeration I have to use:
>			if ((range.commonAncestorContainer.nodeType === 3 &&
>				range.commonAncestorContainer.wholeText === selected) ||
>				(range.commonAncestorContainer.nodeType === 1 &&
>				range.commonAncestorContainer.innerHTML === selected)) {
>				wholeLine = true;
>			}
But that's the current infrastructure limitation and I cant/want change it in order to implement a relatively simple feature.
That's the reason I report the issues in hope that it will be changed one day.

I agree that using plugins is better in that case since it has minimal affect on the core code, but the environment is not ready yet for plugins development as well, so I'll have to hit my head on a wall...